### PR TITLE
[WIP] Task looper

### PR DESCRIPTION
**Pull request by wkrzemien**
Attempt for the iterative task scheme.

Still some unit test examples must be added

### CODE REVIEW

File: Core/JPetTaskLooper/JPetTaskLooper.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ *  @copyright Copyright 2017 The J-PET Framework Authors. All rights reserved.
```
Comment (kkacprzak) [inline Core/JPetTaskLooper/JPetTaskLooper.cpp]: (not that important) maybe 2018?

File: Core/JPetTaskLooper/JPetTaskLooper.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  for (auto subTask : subTasks) {
+    if (!subTask) {
+      ERROR("some subTasks are nullptr");
+      return false;
+    }
+  }
+  while (fIsCondition(inParams)) {
+    for (auto subTask : subTasks) {
+      subTask->init(inParams);
+      subTask->run(nullDataObject);
```
Comment (kkacprzak) [inline Core/JPetTaskLooper/JPetTaskLooper.cpp]: I do not quite get it, why sub tasks run on null data?

File: Core/JPetTaskLooper/JPetTaskLooper.cpp (hunk 1 above)
Comment (wkrzemien) [reply Core/JPetTaskLooper/JPetTaskLooper.cpp]: Ok. Good catch

File: Core/JPetTaskLooper/JPetTaskLooper.cpp (hunk 2 above)
Comment (wkrzemien) [reply Core/JPetTaskLooper/JPetTaskLooper.cpp]: nullDataObject, cause I must pass something - and I didnt want to pass anything special

I will try to add this example + some description also in the code - it is true that it is hard to understand how it suppose to work

Comment (kkacprzak) [reply Core/JPetTaskLooper/JPetTaskLooper.cpp]: Well, it is intended to be universal, so must be a bit abstract. So please ad example/test and description/documentation for doxygen.

**Comment by wkrzemien** [comment]
I messed up with the git repo, so I must clean it up and reopen it later

**Pull request closed without merging**